connectors/filtering/basic_rule.py

```diff
@@ -253,6 +267,23 @@ def default_rule(cls):
         )
 
     def matches(self, document):
+        """Method to check whether a document matches the basic rule.
+
+        A basic rule matches or doesn't match a document based on the following comparisons:
+            - STARTS_WITH: Does the document's field value start with the basic rule's value?
+            - ENDS_WITH: Does the document's field value end with the basic rule's value?
+            - CONTAINS: Does the document's field value contain the basic rule's value?
+            - REGEX: Does the document's field value match the basic rule's regex?
+            - LESS_THAN: Is the document's field value less than the basic rule's value?
+            - GREATER_THAN: Is the document's field value greater than the basic rule's value?
+            - EQUALS: Is the document's field value equal to the basic rule's value?
+
+        If the basic rule is the default rule it's always a match.
+        If the field is not in the document it's always a no match.
+
+        Arguments:
+        - `document`: document to check, if it matches
```

Comment:
```suggestion
        - `document`: document to check for matches
```

Reply:
A document has one match per rule or no match. "for matches" would imply that the doc can be matched multiple for the same rule IMO